src/datastore/migrations.ts

```diff
@@ -99,3 +109,98 @@ export async function dangerousDropAllTables(opts?: {
     await sql.end();
   }
 }
+
+// Function to finish running sql migrations that are too complex for the node-pg-migrate library.
+async function completeSqlMigrations(client: Client, clientConfig: PgClientConfig) {
+  try {
+    await client.query('BEGIN');
+    await complete_1680181889941_contract_log_json(client, clientConfig);
+    await client.query('COMMIT');
+  } catch (error) {
+    await client.query('ROLLBACK');
+    throw error;
+  }
+}
+
+async function* pgCursorQuery<R extends QueryResultRow = any>(args: {
+  clientConfig: PgClientConfig;
+  queryText: string;
+  queryValues?: any[];
+  batchSize: number;
+}) {
+  const cursorClient = new Client(args.clientConfig);
+  try {
+    await cursorClient.connect();
+    const cursor = new PgCursor<R>(args.queryText, args.queryValues);
+    const cursorQuery = cursorClient.query(cursor);
+    let rows: R[] = [];
+    do {
+      rows = await new Promise((resolve, reject) => {
+        cursorQuery.read(args.batchSize, (error, rows) => (error ? reject(error) : resolve(rows)));
+      });
+      yield* rows;
+    } while (rows.length > 0);
+  } finally {
+    await cursorClient.end();
+  }
+}
+
+async function complete_1680181889941_contract_log_json(
```

Comment:
Is there a pressing issue that would require us to ship this feature with this kind of migration? This code looks good to me (and I like this experimentation as you specified in the PR desc), but I worry that it could create some problems if we want to add newer migrations that rely on the new columns or something. IMO it would be easier and more maintainable to just ship this as a regular new migration and rely on replay, or to explicitly mark this code as a "patch" somehow which we could drop once we switch to a new v8 release.

What do you think?